File: spec/mailinator_spec.rb
require File.join(File.dirname(__FILE__), 'spec_helper')

describe Mailinator do
  before(:each) do
    @random_name = "awesomemail#{rand(420)}"
    @primary_email = "#{@random_name}@mailinator.com"
  end

  it 'should have a primary email address' do
    @mail = Mailinator.new(@random_name)
    @mail.email.should == @primary_email
  end

  it 'should have an inbox URL' do
    @mail = Mailinator.new(@random_name)
    @mail.inbox_url.should == "http://mailinator.com/maildir.jsp?email=#{@random_name}"
  end

  it 'should have an RSS URL' do
    @mail = Mailinator.new(@random_name)
    @mail.rss_url.should == "http://mailinator.com/rss.jsp?email=#{@random_name}"
  end

  it 'should have an alternate email address'
  it 'should have an RSS feed'
  it 'should fetch a random primary address'
end

File: javascript/test/spec/mailinator_spec.js
Screw.Unit(function() {

  describe('Mailinator', function() {

    // Setup
    var mail = null;
    var randomName = null;
    var primaryEmail = null;

    before(function() {
      randomName   = "awesomeemail" + Math.floor(Math.random() * 420);
      primaryEmail = randomName + '@mailinator.com';
      mail = new Mailinator(randomName);
    });

    it('should have a primary email address', function() {
      expect(mail.email).to(equal, primaryEmail);
    });

    it('should have an inbox URL', function() {
      inboxURL = 'http://mailinator.com/maildir.jsp?email=' + randomName;
      expect(mail.inboxURL).to(equal, inboxURL);
    });

    it('should have an RSS URL', function() {
      rssURL = 'http://mailinator.com/rss.jsp?email=' + randomName;
      expect(mail.rssURL).to(equal, rssURL);
    });

    it('should have a widget URL', function() {
      width  = '500';
      height = '250';
      widgetURL  = 'http://mailinator.com/widget/mailin8r.jsp';
      widgetURL += '?w=' + width;
      widgetURL += '&h=' + height;
      widgetURL += '&b=' + randomName;
      expect(mail.widgetURL(width, height)).to(equal, widgetURL);
    });

    it('should have an alternate email address', function() {
      expect('pending').to(equal, true);
    });

    it('should have an RSS feed', function() {
      expect('pending').to(equal, true);
    });

    it('should fetch a random primary email address', function() {
      expect('pending').to(equal, true);
    });

  }); // Mailinator

});
